lynxkite-app/web/src/workspace/nodes/Group.tsx

```diff
@@ -1,63 +1,187 @@
       style={{
         width: props.width,
         height: props.height,
         backgroundColor: COLORS[currentColor],
+        position: "relative",
       }}
     >
       <button
-        className="node-group-color-picker-icon"
+        ref={buttonRef}
         onClick={toggleColorPicker}
         aria-label="Change group color"
+        style={{
+          background: "transparent",
+          border: "none",
+          cursor: "pointer",
+          zIndex: 10,
+          transform: "scale(1.25)",
+          transformOrigin: "center",
+        }}
       >
         <Tooltip doc="Change color">
           <Palette />
         </Tooltip>
       </button>
-      {displayingColorPicker && <ColorPicker currentColor={currentColor} onPick={setColor} />}
+
+      {displayingColorPicker && (
+        <ColorPickerPortal
+          anchorRef={buttonRef}
+          currentColor={currentColor}
+          onPick={setColor}
+          onRequestClose={() => setDisplayingColorPicker(false)}
+        />
+      )}
+
+      <NodeResizeControl minWidth={100} minHeight={100}>
+        <ChevronDownRight className="node-resizer" />
+      </NodeResizeControl>
     </div>
   );
 }
 
-function ColorPicker(props: { currentColor: string; onPick: (color: string) => void }) {
-  const colors = Object.keys(COLORS).filter((color) => color !== props.currentColor);
+function ColorPickerPortal({
+  anchorRef,
+  currentColor,
+  onPick,
+  onRequestClose,
+}: {
+  anchorRef: React.RefObject<HTMLElement | null>;
+  currentColor: string;
+  onPick: (c: string) => void;
+  onRequestClose: () => void;
+}) {
+  const portalRef = useRef<HTMLDivElement | null>(null);
+  const [pos, setPos] = useState({ left: 0, top: 0 });
+  const { zoom } = useViewport();
+
+  useEffect(() => {
+    const anchor = anchorRef.current;
+    const portalEl = portalRef.current;
+    if (!anchor || !portalEl) return;
+
+    const update = () => {
+      const aRect = anchor.getBoundingClientRect();
+      const pRect = portalEl.getBoundingClientRect();
+      const margin = 6;
+
+      const left = aRect.right - pRect.width;
+
+      const top = aRect.bottom + margin;
+
+      setPos({ left, top });
+    };
+
+    update();
+    window.addEventListener("scroll", update, true);
+    window.addEventListener("resize", update);
+
+    return () => {
+      window.removeEventListener("scroll", update, true);
+      window.removeEventListener("resize", update);
+    };
+  }, [anchorRef]);
+
+  useEffect(() => {
+    function onDocMouseDown(e: MouseEvent) {
+      const portalEl = portalRef.current;
+      const anchorEl = anchorRef.current;
+      if (portalEl?.contains(e.target as Node)) return;
+      if (anchorEl?.contains(e.target as Node)) return;
+
+      onRequestClose();
+    }
+
+    document.addEventListener("mousedown", onDocMouseDown);
+    return () => document.removeEventListener("mousedown", onDocMouseDown);
+  }, [onRequestClose, anchorRef]);
+
+  return createPortal(
```

Comment:
Interesting! So it can put some stuff into some other location in the DOM. Maybe this is what we need for tooltips?